### PR TITLE
feat: add TypeScript definitions

### PR DESCRIPTION
**Pull request by friedjoff**
And bundle with esbuild.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,16 +36,18 @@
...
     "stylelint": "15.4.0",
     "stylelint-config-standard": "32.0.0",
+    "typescript": "5.0.4",
     "webpack": "4.44.2",
     "webpack-cli": "3.3.12",
     "webpack-dev-server": "3.11.0"
   },
   "scripts": {
-    "build": "webpack --mode production ",
+    "build": "shx rm -rf build && tsc --project config/tsconfig-build.json && esbuild build/index.js --bundle --outfile=build/bundle.js --loader:.svg=text",
```
Comment (oterral) [inline package.json:50]: --minify ?

Comment (oterral) [inline package.json:50]: Is there multiple tsconfig ? if no you can simply call it tsconfig.json and remove the --project parameter

Comment (friedjoff) [reply package.json]: Good point, done.

Comment (friedjoff) [reply package.json]: Not yet, but I guess we should have a default tsconfig.json file for development pretty soon. That's how OL is doing it

**Comment by friedjoff** [comment]
> Nice . I think you can get rid fo webpacks packages using the basic serve functionnality of esbuild to replace webpack-dev-server.
> 
> esbuild build/index.js --bundle --outfile=build/bundle.js --loader:.svg=text --minify --sourcemap --watch --serve

Good idea, I've tried, but Cypress doesn't seem happy yet.

File: package.json [hunk 2]
```diff
@@ -56,17 +51,10 @@
...
     "test": "jest \"src/.*\\.test\\.js\" --passWithNoTests --env=jsdom --transformIgnorePatterns \"/node_modules/(?!(ol|ol-mapbox-style)/).*/\"",
     "test:watch": "yarn test --watchAll"
   },
-  "browserslist": [
-    ">0.2%",
-    "not dead",
-    "not op_mini all",
-    "not ie <= 11",
-    "not android < 5"
-  ],
```
Comment (oterral) [inline package.json:69]: I will leave browserslist, because it's important in  mapset .... or maybe not something to test.

**Comment by oterral** [comment]
> Good idea, I've tried, but Cypress doesn't seem happy yet.

There is always someone unhappy with progress :-D

File: package.json [hunk 3]
```diff
@@ -51,7 +51,7 @@
     "publish:public": "yarn release && yarn build && git push origin HEAD && git push --tags && yarn publish",
     "publish:public:dryrun": "yarn release --dry-run",
     "release": "standard-version",
-    "start": "esbuild src/index.js --bundle --global-name=ole --loader:.svg=dataurl --minify --outfile=index.js --serve --servedir=. --sourcemap --watch",
+    "start": "esbuild src/index.js --bundle --global-name=ole --loader:.svg=dataurl --minify --outfile=index.js --serve --servedir=. --sourcemap --watch=forever",
```
Comment (oterral) [inline package.json:54]: I start often the server my PC will crash :D . Could we do a start:cy target:

"start:cy": yarn start --watch=forever

or maybe using cy:test directly:

"cy:test": "start-server-and-test \"yarn start --watch=forever\" http://127.0.0.1:8000 cy:run",

Comment (oterral) [reply package.json:54]: esbuild will use the last --watch property . Id o that with --outfile property in sev

Comment (friedjoff) [reply package.json]: I've tested running `yarn start` with `--watch=forever`, once I stop the Yarn task all related Node.js process got killed. Can you check if they still run on your machine after stopping the Yarn task?

Comment (oterral) [reply package.json]: yes

Comment (oterral) [reply package.json]: @friedjoff  you re right. it s good for me

File: tasks/prepare-package.mjs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  // write out simplified package.json
+  pkg.main = 'index.js';
+  delete pkg.browserslist;
+  delete pkg.devDependencies;
+  delete pkg.eslintConfig;
+  delete pkg['lint-staged'];
+  delete pkg.prettier;
+  delete pkg.scripts;
+  delete pkg.stylelint;
```
Comment (oterral) [inline tasks/prepare-package.mjs:18]: we could move some config outside package.json . what do you think?

**Comment by oterral** [comment]
@friedjoff  for the node bug on test. Upgrade the .nvmrc file and use it in the action like this:

https://github.com/geops/trafimage-maps/blob/master/.github/workflows/cypress-chrome.yml#L12

File: package.json (hunk 2 above)
Comment (friedjoff) [reply package.json:69]: Do you know how to test this?

**Comment by friedjoff** [comment]
> for the node bug on test. Upgrade the .nvmrc file and use it in the action like this:

For apps I agree with using something like `.nvmrc`, but for open source libraries we should not pin to a specific Node.js version to catch incompatibilities with different versions. I've changed the matrix strategy for our Cypress GitHub Action to use the oldest maintained LTS release and the latest stable release to make sure we support a sensible range of Node.js releases.

**Comment by oterral** [comment]
>  but for open source libraries we should not pin to a specific Node.js version to catch incompatibilities with different versions

If you pin everybody will use the same  node version so no more problem.  Dependency like node-canvas are already node version dependent.  The only problem I had with node version is when they were not pinned so CI didn't use the good node version and test failed for a month, and nobody care.

**Comment by oterral** [comment]
Very nice thanks

**Pull request merged**